File: 0383. Ransom Note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] count = new int[26];
        for(char ch: magazine.toCharArray()) count[ch - 'a']++;

        for(char ch: ransomNote.toCharArray()){
            if(--count[ch - 'a'] < 0) return false;
        }
        return true;
    }
}

File: 0383. Ransom Note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        freq = {}
        for c in magazine:
            freq[c] = freq.get(c, 0) + 1

        for c in ransomNote:
            freq[c] = freq.get(c, 0) - 1
            if freq[c] < 0:
                return False
        
        return True
